### PR TITLE
[MM-53113] cmd/mattermost/db: remove app dependency from db reset command

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Avoid starting the `app.App` to access store. We now follow a similar pattern in other db commands such as migrate.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53113

#### Release Note

```release-note
mattermost db reset command no longer starts the application server, it will only start the store layer and truncate the tables excluding the migrations table.
```

### CODE REVIEW

**Review by agnivade — APPROVED** [review]

**Review by stafot — APPROVED** [review]
lgtm

File: server/cmd/mattermost/commands/db.go [hunk 1]
```diff
@@ -124,11 +122,11 @@ func initDbCmdF(command *cobra.Command, _ []string) error {
 }
 
 func resetCmdF(command *cobra.Command, args []string) error {
-	a, err := InitDBCommandContextCobra(command, app.SkipPostInitialization())
+	cfgDSN := getConfigDSN(command, config.GetEnvironment())
```
Comment (mgdelacroix) [inline server/cmd/mattermost/commands/db.go:125]: Not blocking the PR with this, but if we're going to follow this same pattern for other commands, I think we could refactor this into a helper similar to the `InitDBCommandContextCobra` that returns a store initialized from the config.

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1:

File: server/cmd/mattermost/commands/db.go [hunk 2]
```diff
@@ -146,11 +144,11 @@ func resetCmdF(command *cobra.Command, args []string) error {
 		}
 	}
 
-	a.Srv().Store().DropAllTables()
-	CommandPrettyPrintln("Database successfully reset")
+	config := cfgStore.Get()
+	ss := sqlstore.New(config.SqlSettings, nil)
```
[the remark this answers is not in the document]
Comment (isacikgoz) [reply server/cmd/mattermost/commands/db.go]: good catch, added!

File: server/cmd/mattermost/commands/db.go (hunk 1 above)
Comment (isacikgoz) [reply server/cmd/mattermost/commands/db.go]: In fact, we can just use command for that similar to `InitDBCommandContextCobra`, added the suggestion.

**Comment by cwarnermm** [comment]
@isacikgoz - Is there any outstanding documentation effort needed for this PR from your perspective?

**Comment by isacikgoz** [comment]
@cwarnermm Ideally not, the release note should be fine. I don't think anyone using the `db reset` command to rely on app as it removes all the data from the database 🤞

**Pull request merged**